src/App.tsx

```diff
@@ -16,35 +22,101 @@
 ];
 
 export const App: React.FC = () => {
+  const [sort, setSort] = useState<string[]>([...goodsFromServer]);
+  const [reverse, setReverse] = useState<boolean>(false);
+  const [active, setActive] = useState<string>('');
+
+  const handleSort = (good: Cases): void => {
+    let newSort: string[];
+
+    switch (good) {
+      case Cases.Abc:
+        if (reverse) {
+          newSort = [...sort].sort((a, b) => b.localeCompare(a));
+          setActive(good);
+        } else {
+          newSort = [...sort].sort((a, b) => a.localeCompare(b));
+          setActive(good);
+        }
+
+        break;
+      case Cases.Length:
+        if (reverse) {
+          newSort = [...sort].sort((a, b) => b.length - a.length);
+          setActive(good);
+        } else {
+          newSort = [...sort].sort((a, b) => a.length - b.length);
+          setActive(good);
+        }
+
+        break;
+      case Cases.Reverse:
+        if (active === Cases.Abc || active === Cases.Length) {
+          setReverse(!reverse);
+        } else if (active === Cases.Reverse) {
+          setReverse(!reverse);
+          setActive('');
+        } else {
+          setReverse(!reverse);
+          setActive(Cases.Reverse);
```

Comment:
The logic for toggling the reverse state and setting the active state is a bit convoluted. Consider simplifying this logic for better readability and maintainability.